x-pack/plugins/fleet/public/applications/fleet/sections/settings/components/edit_output_flyout/use_output_form.tsx

```diff
@@ -311,35 +578,132 @@ export function useOutputForm(onSucess: () => void, output?: Output) {
       }
 
       const proxyIdValue = proxyIdInput.value !== '' ? proxyIdInput.value : null;
-      const data: PostOutputRequest['body'] = isLogstash
-        ? {
-            name: nameInput.value,
-            type: typeInput.value as 'elasticsearch' | 'logstash',
-            hosts: logstashHostsInput.value,
-            is_default: defaultOutputInput.value,
-            is_default_monitoring: defaultMonitoringOutputInput.value,
-            config_yaml: additionalYamlConfigInput.value,
-            ssl: {
-              certificate: sslCertificateInput.value,
-              key: sslKeyInput.value,
-              certificate_authorities: sslCertificateAuthoritiesInput.value.filter(
-                (val) => val !== ''
-              ),
-            },
-            proxy_id: proxyIdValue,
-            ...shipperParams,
+
+      const payload: NewOutput = (() => {
+        const parseIntegerIfStringDefined = (value: string | undefined): number | undefined => {
+          if (value !== undefined) {
+            const parsedInt = parseInt(value, 10); // Specify the base as 10 for decimal numbers
+            if (!isNaN(parsedInt)) {
+              return parsedInt;
+            }
           }
-        : {
-            name: nameInput.value,
-            type: typeInput.value as 'elasticsearch' | 'logstash',
-            hosts: elasticsearchUrlInput.value,
-            is_default: defaultOutputInput.value,
-            is_default_monitoring: defaultMonitoringOutputInput.value,
-            config_yaml: additionalYamlConfigInput.value,
-            ca_trusted_fingerprint: caTrustedFingerprintInput.value,
-            proxy_id: proxyIdValue,
-            ...shipperParams,
-          };
+          return undefined;
+        };
+
+        switch (typeInput.value) {
+          case outputType.Kafka:
```

Comment:
We could consider splitting this output form to separate components, as there is a lot of fields specific to Kafka.